packages/client/test/events.spec.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  OpenFeature,
+  ApiEvents,
+  ProviderEvents,
+  Provider,
+} from '../src';
+import EventEmitter from 'events';
+
+const ERROR_REASON = 'error';
+const ERROR_CODE = 'MOCKED_ERROR';
+
+const MOCK_EMITTER: EventEmitter = {
+  emit: jest.fn( () => {
+    return true;
+  }),
+  initialize: jest.fn( () => {
+    return MOCK_EMITTER;
+  })
+}as unknown as EventEmitter;
+
+const MOCK_ERROR_EMITTER: EventEmitter = {
+  emit: jest.fn( (event) => {
+    if (event == ProviderEvents.Ready) {
+      throw Error('Event error!');
+    }
+    return true;
+  }),
+  initialize: jest.fn( () => {
+    return MOCK_EMITTER;
+  })
+}as unknown as EventEmitter;
+
+const MOCK_PROVIDER: Provider = {
+  metadata: {
+    name: 'mock-events-success',
+  },
+  initialize: jest.fn(() =>  {
+    return Promise.resolve(undefined);
+  }),
+  events: MOCK_EMITTER,
+} as unknown as Provider;
+
+const ERROR_PROVIDER: Provider = {
+  metadata: {
+    name: 'mock-events-failure',
+  },
+  initialize: jest.fn(() =>  {
+    return Promise.resolve(undefined);
+  }),
+  events: MOCK_ERROR_EMITTER,
+} as unknown as Provider;
+
+describe('Events', () => {
+  // set timeouts short for this suite.
+  jest.setTimeout(1000);
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('Requirement 5.1.1', () => {
+    it('The provider **MAY** define a mechanism for signalling the occurrence of an event`PROVIDER_READY`',
+       async () => {
+
+         await OpenFeature.setProvider(MOCK_PROVIDER);
+         expect(OpenFeature['_provider']).toBe(MOCK_PROVIDER);
```

Comment:
I don't think we need to test this private prop. If you want, you can test the metadata: https://github.com/open-feature/js-sdk/blob/0c2befdbdbfbf46dede010e00f0efcd8e0f0f160/packages/client/src/open-feature.ts#L57